File: core/src/org/tetawex/ecf/util/Bundle.java
package org.tetawex.ecf.util;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by tetawex on 28.01.17.
 */
public class Bundle {
    private Map<String, Object> items;

    public Bundle() {
        items = new HashMap<String, Object>();
    }

    public void putItem(String key, Object item) {
        items.put(key, item);
    }

    public <T> T getItem(String key, Class<T> c) {
        return c.cast(items.get(key));
    }

    public <T> T getItem(String key, Class<T> c, T defaultValue) {
        if (items.containsKey(key))
            return c.cast(items.get(key));
        return defaultValue;
    }
}

File: core/src/org/tetawex/ecf/util/Bundle.kt
package org.tetawex.ecf.util

import java.util.HashMap

/**
 * Created by tetawex on 28.01.17.
 */
class Bundle {
    private val items: MutableMap<String, Any>

    init {
        items = HashMap()
    }

    fun putItem(key: String, item: Any) {
        items[key] = item
    }

    fun <T> getItem(key: String, c: Class<T>): T? {
        return c.cast(items[key])
    }

    fun <T> getItem(key: String, c: Class<T>, defaultValue: T): T? {
        return if (items.containsKey(key)) c.cast(items[key]) else defaultValue
    }
}
